pkg/scheduler/allocator.go

```diff
@@ -203,20 +208,14 @@ func (m *Scheduler) tryBatchAllocation(partition string, partitionContext *Parti
     return allocations[:allocatedLength], failedAsks[:failedAskLength]
 }
 
-// TODO: convert this as an interface.
-func (m *Scheduler) evaluateForSchedulingPolicy(nodes []*SchedulingNode, partitionContext *PartitionSchedulingContext) NodeIterator {
-    // Sort Nodes based on the policy configured.
-    configuredPolicy:= partitionContext.partition.GetNodeSortingPolicy()
-    switch configuredPolicy {
-    case common.BinPackingPolicy:
-        SortNodes(nodes, MinAvailableResources)
-        return NewDefaultNodeIterator(nodes)
+func (m *Scheduler) getNodeSortType(partitionSortingPolicy common.SortingPolicy) SortType {
+    switch partitionSortingPolicy {
     case common.FairnessPolicy:
-        SortNodes(nodes, MaxAvailableResources)
-        return NewDefaultNodeIterator(nodes)
+        return MaxAvailableResources
+    case common.BinPackingPolicy:
+        return MinAvailableResources
     }
-
-    return nil
+    return MaxAvailableResources
 }
```

Comment:
I have tried to understand why we need sort type for nodes when we already have sort policy, and I thought maybe it's for flexibility, for example, we may use MaxAvailableRatio instead of MaxAvailableResources for FairnessPolicy. Will this be considered later?

Reply:
No, I don't think we need that. the case you mentioned can be just 2 different policies.

Reply:
OK